src/app.js

```diff
@@ -39,7 +40,12 @@ const APP_VIEWS = [
   path.resolve("node_modules/@govuk-one-login")
 ];
 
-app.set("view engine", configureNunjucks(app, APP_VIEWS));
+app.use((req, res, next) => {
```

Comment:
If you use `currentUrl` as a locals variable (see line 81) and use it as a parameter when you call your nunjuck filter, you will not need to ask to other frontend apps to add this additional piece of code.
That will give you something like that (just an example, don't reuse it without testing):
line: 82 -> `res.locals.currentUrl =  req.protocol + "://" + req.get("host") + req.originalUrl;`
And when you call your new nunjuck filter -> `{{ addLanguageParam(lang.code, currentUrl)}}`

I think this solution will simplify the language toggle implementation.